src/BaseOfTalents/WebUI/Auth/Services/UserAccountService.cs

```diff
@@ -0,0 +1,191 @@
+        {
+            var user = _userService.Get(login);
+            if (user == null && user.Password.Equals(password))
+            {
+                throw new ArgumentException("Wrong login or password");
+                //TODO: Extract message to external source
+                //TODO: new exception type
+            }
+            return user;
+        }
+
+        public bool CkeckAuthority(string login, string password)
+        {
+            try
+            {
+                Authentificate(login, password);
+                return true;
+            }
+            catch (ArgumentException)
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Perfoms accessing to user of specified login and password async
+        /// </summary>
+        /// <param name="login">Application user login</param>
+        /// <param name="password">User password (hashed)</param>
+        /// <returns>Corresponting user dto object</returns>
+        /// <exception cref="ArgumentException">Is thrown, when there is no user with such a login and password</exception>
+        public async Task<UserDTO> AuthentificateAsync(string login, string password)
+        {
+            var user = await _userService.GetAsync(login);
+            if (user == null && user.Password.Equals(password))
+            {
+                throw new ArgumentException("Wrong login or password");
+                //TODO: Extract message to external source
+                //TODO: new exception type
+            }
+            return user;
+        }
+
+        public async Task<bool> CkeckAuthorityAsync(string login, string password)
+        {
+            try
+            {
+                await AuthentificateAsync(login, password);
+                return true;
+            }
+            catch (ArgumentException)
+            {
+                return false;
+            }
+        }
+
+
+        /// <summary>
+        /// Recovering account by email or login
+        /// </summary>
+        /// <param name="loginOrEmail">string contains user's login or email</param>
+        public void RecoverAccount(string loginOrEmail)
+        {
+            if (String.IsNullOrEmpty(loginOrEmail))
+            {
+                throw new ArgumentException("Login and email can not be empty!");
+            }
+
+            var _validator = new EmailStringValidator();
+            UserDTO _user = _userService.Get((usr) => _validator.IsEmail(loginOrEmail) ?
+                                                              usr.Email == loginOrEmail :
+                                                              usr.Login == loginOrEmail);
+
+            if (_user == null)
+            {
+                throw new ObjectNotFoundException("User with such login or email not found!");
+            }
+
+            string pwd = PasswordGenerator.GeneratePassword();
+            _user.Password = (Password)pwd;
```

Comment:
#duplicate
you can use constructor of Password
may be it be better if PasswordGenerator.GeneratePassword will return Password object?